Workaround for nvc bug that incorrectly pads out c99 arrays with gaps
Ref. https://forums.developer.nvidia.com/t/incorrect-code-for-statically-initialized-arrays/189934

Author: micketeer@gmail.com
diff -ru openmpi-4.1.1.orig/opal/datatype/opal_datatype_module.c openmpi-4.1.1/opal/datatype/opal_datatype_module.c
--- openmpi-4.1.1.orig/opal/datatype/opal_datatype_module.c	2021-09-23 19:23:25.812038609 +0200
+++ openmpi-4.1.1/opal/datatype/opal_datatype_module.c	2021-09-23 19:25:47.306649251 +0200
@@ -88,6 +88,10 @@
  */
 OPAL_DECLSPEC const size_t opal_datatype_local_sizes[OPAL_DATATYPE_MAX_PREDEFINED] =
 {
+    [OPAL_DATATYPE_LOOP] = 0,
+    [OPAL_DATATYPE_END_LOOP] = 0,
+    [OPAL_DATATYPE_LB] = 0,
+    [OPAL_DATATYPE_UB] = 0,
     [OPAL_DATATYPE_INT1] = sizeof(int8_t),
     [OPAL_DATATYPE_INT2] = sizeof(int16_t),
     [OPAL_DATATYPE_INT4] = sizeof(int32_t),
